### ISSUE TITLE
Please give a tip why it's always in 45 FPS?

### ISSUE THREAD

**Issue by yushen0403** [ValveSoftware/openvr#175]
I'm working on custom driver, just launched the SteamVR with it, only a background displayed in HMD, everything worked fine, but just in 45FPS. Any suggestion?

**Comment by JoeLudwig** [comment]
Do you have "Enable Always-on Reprojection" checked in the developer settings? Reducing the frame rate to 45Hz is what that setting does.

**Comment by yushen0403** [comment]
Thanks for your reply, nope, never checked that, but i'm running in SteamVR Beta, does it matter? I will have a try on the sample driver, becuase it's really simple so i can find if it's the problem of driver itself or the code i did.

**Comment by yushen0403** [comment]
The sample driver is also in 45FPS, i found something in VRCompositor.txt, "Forcing debug mode for sample driver", is that the problem?

Fri Jul 01 2016 10:49:19.014 - //=====================================================================================================================
Fri Jul 01 2016 10:49:19.014 - VR compositor (v1467333449) starting up
Fri Jul 01 2016 10:49:19.014 - RuntimePath : C:\Program Files (x86)\Steam\steamapps\common\SteamVR\
Fri Jul 01 2016 10:49:19.244 - Forcing debug mode for sample driver.
Fri Jul 01 2016 10:49:19.244 - Events created
Fri Jul 01 2016 10:49:19.264 - Initializing D3D resources...
Fri Jul 01 2016 10:49:19.507 - Headset is using direct mode
Fri Jul 01 2016 10:49:19.662 - LiquidVR: Driver version 16.150.2401.0
Fri Jul 01 2016 10:49:19.666 - LiquidVR: RenderWait functionality initialized successfully.
Fri Jul 01 2016 10:49:19.672 - Creating liquidvr display
Fri Jul 01 2016 10:49:19.672 - Loading shaders
Fri Jul 01 2016 10:49:19.697 - Creating samplers
Fri Jul 01 2016 10:49:19.697 - Creating util resources
Fri Jul 01 2016 10:49:19.698 - Creating distortion surfaces
Fri Jul 01 2016 10:49:19.774 - Creating constant buffers
Fri Jul 01 2016 10:49:19.774 - Creating auxiliary resources
Fri Jul 01 2016 10:49:19.774 - Loading mct resources
Fri Jul 01 2016 10:49:19.774 - Initializing timing resources
Fri Jul 01 2016 10:49:19.778 - Distort initialized
Fri Jul 01 2016 10:49:19.792 - Renderer started
Fri Jul 01 2016 10:49:19.792 - Startup Complete (2.357568 seconds)
Fri Jul 01 2016 10:49:19.793 - Found 4 quads for Collision Bounds
Fri Jul 01 2016 10:49:19.793 - Found Collision Bounds info m_pChaperone->GetCollisionBoundsInfo( nQuadCount = 4 )
Fri Jul 01 2016 10:49:19.793 -    Collision Bounds 0 = -0.500000, 0.000000, -0.500000
Fri Jul 01 2016 10:49:19.793 -    Collision Bounds 1 = -0.500000, 0.000000, 0.500000
Fri Jul 01 2016 10:49:19.793 -    Collision Bounds 2 = 0.500000, 0.000000, 0.500000
Fri Jul 01 2016 10:49:19.793 -    Collision Bounds 3 = 0.500000, 0.000000, -0.500000
Fri Jul 01 2016 10:49:19.793 - Collision Bounds appear to be standing bounds, so render a persistent circle instead
Fri Jul 01 2016 10:49:19.793 - Creating solid play area quad for compositor construct: 1.00 x 1.00
Fri Jul 01 2016 10:49:19.796 - AUDIO: AudioDeviceChangeNotice (noting refresh): 2, device = {0.0.0.00000000}.{bb916d57-b5ca-40fe-8406-0863330b87b4}
Fri Jul 01 2016 10:49:19.796 - AUDIO: OnDeviceStateChanged: DEVICE_STATE_ACTIVE (0x00000001), device = {0.0.0.00000000}.{bb916d57-b5ca-40fe-8406-0863330b87b4}
Fri Jul 01 2016 10:49:19.864 - Found 4 quads for Collision Bounds
Fri Jul 01 2016 10:49:19.864 - Found Collision Bounds info m_pChaperone->GetCollisionBoundsInfo( nQuadCount = 4 )
Fri Jul 01 2016 10:49:19.864 -    Collision Bounds 0 = -0.500000, 0.000000, -0.500000
Fri Jul 01 2016 10:49:19.864 -    Collision Bounds 1 = -0.500000, 0.000000, 0.500000
Fri Jul 01 2016 10:49:19.864 -    Collision Bounds 2 = 0.500000, 0.000000, 0.500000
Fri Jul 01 2016 10:49:19.864 -    Collision Bounds 3 = 0.500000, 0.000000, -0.500000
Fri Jul 01 2016 10:49:19.864 - Collision Bounds appear to be standing bounds, so render a persistent circle instead

**Comment by JoeLudwig** [comment]
Renaming the sample driver to something else might help. The name "sample" kicks the compositor into debug mode where it expects to show in a window on the desktop instead of an actual HMD. That turns off full screen and limits framerate to 60Hz.

**Comment by yushen0403** [comment]
yep, i just found that, it's forced in debug mode so i did like you suggested. the background and tutorial is also in 45 FPS, but a heavy game in 90 FPS :), really weird. so that's not a real problem i think.

thanks for your reply JoeLudwig.

**Comment by JoeLudwig** [comment]
What kind of hardware are you running on? Maybe it's not keeping up?

**Comment by yushen0403** [comment]
Intel Core i7-6700 3.4GHz (8 CPUs), AMD Radeon R9 Fury Series, 16GB Memory, Windows 10 Pro. I played on my computer with HTC Vive, it was really good for each games.

**Comment by yushen0403** [comment]
I got another problem i think, even if it's in 90 FPS, the game still present lagging while headset rotating or moving. i tried if the headset is just put on table, only navigate in game scene with keyboard, it will display very smothly. So i guess it's something about prediction or wrap, but i didn't see anything can help in driver interface except vr::DriverPose_t::poseTimeOffset, and VsyncEvent(), speaking about vsync, i can't get such a time if i was not in charge of rendering, right? do you have any suggestion about this?

**Comment by JoeLudwig** [comment]
It sounds like you're describing what happens when the driver is passing 0 for velocity and angular velocity. That prevents the rest of the system from doing prediction so the pose provided to applications is always the latest pose from the driver (which is always wrong.)  Are you providing velocity and acceleration from your driver?

**Comment by yushen0403** [comment]
yep, i'm going to pr4oviding that.
